### PR TITLE
[documentation] Adding a summary of the specification types available in Liquid Haskell

### PR DESCRIPTION
**Pull request by plredmond**
Prompted by [discussion in slack](https://liquidhaskell.slack.com/archives/C54QAL9RR/p1601612338073600), I wrote this brief summary of the LH annotations I've learned about. I've tried to explain very briefly what they do to the LH internal state, and the constraints on their use. Per @ranjitjhala I've added it here to the documentation. Perhaps though it would be better as a separate page? I'm unsure. Please let me know what you'd prefer in your review comments!

Things to consider:
* Where should this documentation go?
* Is this documentation correct? (I'm still an LH beginner and probably have got things wrong in places)
* Are there additional annotations which should be added to this list?
* Is this the best way to present this information? What improvements can be made?

### CODE REVIEW

File: docs/mkDocs/docs/specifications.md [hunk 1]
```diff
@@ -1,7 +1,46 @@
...
+* `{-@ reflect <function-name> @-}` creates an uninterpreted function of the same name in the refinement logic,
+  copies the implementation to a refinement type alias,
+  and adds a refinement to the type of the uninterpreted function that specifies the type alias as a post-condition.
+  ([See more: Section 2.2 of this paper](http://goto.ucsd.edu/~nvazou/refinement-reflection/refinement-reflection.pdf))
+    * All parts of the definition must already be available to the refinement logic.
+    * The function may be recursive.
+* `{-@ type <type-alias-head> = <refinement-type> @-}` introduces a type alias that looks like Haskell syntax but can contain refinements and may be parameterized over both types and values.
+  ([Jump to: Type Aliases](#type-aliases))
+* `{-@ predicate .. @-}` introduces something like `{-@ type .. @-}`.
+  (_Deprecated_, [Jump to: Predicate Aliases](#predicate-aliases))
```
Comment (ranjitjhala) [inline docs/mkDocs/docs/specifications.md]: maybe add that its deprecated in favor of using `inline` ?

**Comment by ranjitjhala** [comment]
This is awesome, lmk if you want to make the minor edit above, else would love to merge, thanks!

**Comment by plredmond** [comment]
Yup! I'll make the edit. I just didn't see my email until now.

File: docs/mkDocs/docs/specifications.md (hunk 1 above)
Comment (plredmond) [inline docs/mkDocs/docs/specifications.md]:
```suggestion
  (_Deprecated, use `inline` instead_, [Jump to: Predicate Aliases](#predicate-aliases))
```

**Comment by plredmond** [comment]
@ranjitjhala let me know if you'd like other changes! i should be quicker than over the weekend :)

**Comment by ranjitjhala** [comment]
Thanks @plredmond -- this is awesome, lets merge, we can always make it better going forward!

**Pull request merged**